index.js

```diff
@@ -1,20 +1,39 @@
 const util = require('util');
+const dailyVersion = require('daily-version');
 const execFile = util.promisify(require('child_process').execFile);
 
 async function init() {
-	await execFile('git', ['fetch', '--tags']);
-	const {stdout: tagOnHead} = await execFile('git', ['tag', '-l', '--points-at', 'HEAD']);
-	if (tagOnHead) {
-		console.log('::set-output name=version::' + tagOnHead);
-		console.log('No new commits since the last tag');
+	// Use ENV if it's a `push.tag` event
+	if (process.env.GITHUB_REF.startsWith('refs/tags/')) {
+		const pushedTag = process.env.GITHUB_REF.replace('refs/tags/', '');
+		console.log('::set-output name=version::' + process.env.GITHUB_REF.replace('refs/tags/', ''));
+		console.log('Run triggered by tag `' + pushedTag + '`. No new tags will be created by `daily-version-action`.');
 		return;
 	}
 
-	const {stdout: version} = await execFile('npx', ['daily-version']);
+	// Look for tags on the current commit
+	await execFile('git', ['fetch', '--depth=1', 'origin', 'refs/tags/*:refs/tags/*']);
```

Comment:
Does `execFile` need `encoding` parameter similar to https://github.com/fregante/daily-version/commit/fa34122396040218fc2b31edbb2cb1888a1fce29#diff-168726dbe96b3ce427e7fedce31bb0bcR9-R11?

Reply:
For whatever reason, [`execFile`](https://nodejs.org/api/child_process.html#child_process_child_process_execfile_file_args_options_callback) defaults to `string` while [`execFileSync`](https://nodejs.org/api/child_process.html#child_process_child_process_execfilesync_file_args_options) uses `Buffer` 🤷‍♂️